lib/Resque/Worker.php

```diff
@@ -454,7 +460,14 @@ public function pruneDeadWorkers()
 	public function workerPids()
 	{
 		$pids = array();
-		exec('ps -A -o pid,command | grep [r]esque', $cmdOutput);
+
+		if ('SunOS' == PHP_OS) {
+			$cmd = 'ps -A -o pid,comm';
+		} else {
+			$cmd = 'ps -A -o pid,command';
+		}
+
```

Comment:
As discussed in https://github.com/chrisboulton/php-resque/pull/139#issuecomment-26745737, it would probably be better to use `ps -A -o pid,args` here instead of switching commands based on platform...